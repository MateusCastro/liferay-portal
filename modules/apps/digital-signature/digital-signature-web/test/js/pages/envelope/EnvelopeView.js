/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {act, cleanup, render} from '@testing-library/react';
import React, {createContext} from 'react';

import {AppContext} from '../../../../src/main/resources/META-INF/resources/js/AppContext';
import EnvelopeView from '../../../../src/main/resources/META-INF/resources/js/pages/envelope/EnvelopeView';

const data = {
	envelope: {
		createdLocalDateTime: '2021-01-01T12:30:00.000',
		documents: [
			{
				documentId: 'document1',
				fileExtension: 'pdf',
				name: 'file.pdf',
			},
		],
		emailBlurb: 'Email Message',
		emailSubject: 'Email Subject',
		envelopeId: 'envelope1',
		name: 'Test Envelope Name',
		recipients: {
			signers: [
				{
					email: 'test@liferay.com',
					name: 'Recipient Full Name',
					recipientId: 'recipient1',
					status: 'autoresponded',
				},
			],
		},
		senderEmailAddress: 'test@liferay.com',
		status: 'sent',
	},
	fileEntries: [],
};

const Context = createContext();

const EnvelopeViewWithProvider = (props) => (
	<AppContext.Provider
		value={{
			baseResourceURL:
				'http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_digital_signature_web_internal_portlet_DigitalSignaturePortlet',
		}}
	>
		<EnvelopeView {...props} />
	</AppContext.Provider>
);

describe('EnvelopeView', () => {
	afterEach(cleanup);

	beforeEach(() => {
		jest.useFakeTimers();
	});

	it('renders', async () => {
		fetch.mockResponseOnce(JSON.stringify(data));

		const {asFragment} = render(
			<EnvelopeViewWithProvider
				AppContext={Context}
				match={{params: {envelopeId: 1}}}
			/>
		);

		await act(async () => {
			await jest.runAllTimers();
		});

		expect(asFragment()).toMatchSnapshot();
	});
});
