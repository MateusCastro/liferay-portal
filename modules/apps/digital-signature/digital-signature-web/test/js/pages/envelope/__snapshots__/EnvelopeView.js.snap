// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvelopeView renders 1`] = `
<DocumentFragment>
  <div
    class="envelope-view"
  >
    <div
      class="envelope-view__header"
    >
      <div>
        <span
          class="envelope-view__header__title"
        >
          Email Subject
        </span>
        <span
          class="label ml-2 label-primary"
        >
          <span
            class="label-item label-item-expand"
          >
            sent
          </span>
        </span>
      </div>
      <button
        class="btn btn-primary"
        type="button"
      >
        <svg
          class="lexicon-icon lexicon-icon-download"
          role="presentation"
        >
          <use
            xlink:href="#download"
          />
        </svg>
        <span
          class="ml-1"
        >
          download
        </span>
      </button>
    </div>
    <div
      class="taglib-empty-result-message"
    >
      <div
        class="text-center"
      >
        <div
          class="taglib-empty-state mb-2 mt-4"
        />
        <h1
          class="taglib-empty-result-message-title"
        >
          no-preview-available
        </h1>
        <p
          class="empty-message-color taglib-empty-result-message-description"
        >
          the-envelope-does-not-have-a-document-to-preview
        </p>
      </div>
    </div>
    <div
      class="envelope-view__details"
    >
      <div>
        <b>
          envelope-id
        </b>
        : 1
      </div>
      <hr />
      <div
        class="d-flex"
      >
        <div
          class="flex-grow-1"
        >
          <b>
            Email Subject
          </b>
          <span
            class="ml-1"
          />
        </div>
        <div>
          <b>
            1/1/2021 12:30 PM
          </b>
          <span
            class="ml-1"
          />
        </div>
      </div>
      <div>
        <b>
          to:
        </b>
        <span
          class="ml-1"
        >
          test@liferay.com
        </span>
      </div>
      <div>
        <b>
          from:
        </b>
        <span
          class="ml-1"
        >
          test@liferay.com
        </span>
      </div>
      <p
        class="mt-2"
      >
        Email Message
      </p>
    </div>
    <input
      type="hidden"
      value="http://localhost:8080/?p_p_id=unitTest&_unitTest_dsEnvelopeId=1&_unitTest_javax.portlet.action=%2Fdigital_signature%2Fdelete_ds_envelope&p_auth=default-mocked-auth-token&p_p_lifecycle=1"
    />
  </div>
</DocumentFragment>
`;
